src/Turbo/Stream/AddTurboStreamFormatSubscriber.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Kévin Dunglas <kevin@dunglas.fr>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Turbo\Stream;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Detects if it's a Turbo Stream request and set the format accordingly.
+ *
+ * @author Kévin Dunglas <kevin@dunglas.fr>
+ *
+ * @experimental
+ */
+final class AddTurboStreamFormatSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @see https://github.com/hotwired/turbo/issues/24 Explanation of why this hack is necessary
```

Comment:
Hasn't this been fixed?

Reply:
Yes, but this code also uses the new media type that was added, so the comment needs updating.

in addition this is probably only necessary, because Symfony Framework does not have `$request->setFormat(TurboStreamResponse::STREAM_FORMAT, TurboStreamResponse::STREAM_MEDIA_TYPE);` in core yet.

Reply:
Yes indeed, but I think that we should wait for Turbo to be marked stable before adding this media type to `symfony/symfony` (the media type already changed).